tests/utils.test.ts

```diff
@@ -177,4 +179,103 @@ describe("Utils", () => {
       expect(result.length).toBe(0);
     });
   });
+
+  describe("filterOutPastAnnouncements", () => {
+    it("should filter announcements to only include those with an end date in the future", () => {
+      const result = filterFutureAnnouncements(
+        announcements,
+        new Date(2024, 11, 1)
+      );
+      expect(result.length).toBe(1);
+      expect(result[0].id).toBe("3");
+    });
+    it("should filter announcements to only include those with an end date in the future (all announcements have concluded)", () => {
+      const result = filterFutureAnnouncements(
+        duplicateStartAnnouncements,
+        new Date(2026, 11, 1)
+      );
+      expect(result.length).toBe(0);
+    });
+  });
+
+  describe("dateInRange", () => {
+    it("should return true as first date is in the range set by the second two dates", () => {
+      const result = dateInRange(
+        new Date("2024-07-15T14:30:00"),
+        new Date("2024-07-01T14:30:00"),
+        new Date("2024-07-30T14:30:00")
+      );
+      expect(result).toBe(true);
+    });
+    it("the first date is before the range of the second two dates (not in range)", () => {
+      const result = dateInRange(
+        new Date("2024-07-01T12:30:00"),
+        new Date("2024-07-01T14:30:00"),
+        new Date("2024-07-30T14:30:00")
+      );
+      expect(result).toBe(false);
+    });
+    it("the first date is after the range of the second two dates (not in range)", () => {
+      const result = dateInRange(
+        new Date("2024-07-30T16:30:00"),
+        new Date("2024-07-01T14:30:00"),
+        new Date("2024-07-30T14:30:00")
+      );
+      expect(result).toBe(false);
+    });
+    it("the given date is exactly the same date as the earlier bound of the range (in range)", () => {
+      const result = dateInRange(
+        new Date("2024-07-01T14:30:00"),
+        new Date("2024-07-01T14:30:00"),
+        new Date("2024-07-30T14:30:00")
+      );
+      expect(result).toBe(true);
+    });
+    it("the given date is exactly the same date as the later bound of the range (in range)", () => {
+      const result = dateInRange(
+        new Date("2024-07-30T14:30:00"),
+        new Date("2024-07-01T14:30:00"),
+        new Date("2024-07-30T14:30:00")
+      );
+      expect(result).toBe(true);
+    });
+  });
+
+  describe("formatDate", () => {
+    it("should format the date correctly for a date in the middle of the year", () => {
+      const date = new Date("2024-07-15T14:30:00");
+      const result = formatDate(date);
+      expect(result).toBe("7/15 02:30 PM");
```

Comment:
Can you edit the function to remove the `0` before the `2`? I believe the design has it as `2:30` format.